### ISSUE TITLE
IInstalling packages in R-studio Version 1.1.463 on Ubuntu 16.04  needs this fix to install Armadillo

### ISSUE THREAD

**Issue by sgmshaik** [rcppsmc/rcppsmc#47]
The RcppArmadillo, package wont install in the latest R you need to use to following command as a hack . 

 install.packages("https://cran.r-project.org/src/contrib/Archive/RcppArmadillo/RcppArmadillo_0.6.100.0.0.tar.gz", repos=NULL, type="source")

**Comment by eddelbuettel** [comment]
Can you explain what are you talking about? 

What OS?  What version of tools?  As per the CRAN results page for both package, either installs everywhere:

https://cloud.r-project.org/web/checks/check_results_RcppSMC.html
https://cloud.r-project.org/web/checks/check_results_RcppArmadillo.html

Please substantiate, or close.

**Comment by sgmshaik** [comment]
Hi EddeLBuettell, It's for version RStudio Version 1.1.463. 

It took me a while to figure out the command and I thought it might be useful for someone else.

**Comment by eddelbuettel** [comment]
Utter nonsense.  For starters Ubuntu also has a r-cran-rcpparmadillo package I maintain.

If you are confused about anything ask on the r-sig-debian list but please don't file random bugs.

**Comment by eddelbuettel** [comment]
Also, my name is Dirk.  My family name is Eddelbuettel.  It is consider rude to address people by their family name, and more so to not even care to write it correctly.

**Comment by sgmshaik** [comment]
Dear Dirk, I just downloaded both R-base and R-studio, and it no longer installs Armadillo. I was just trying to help out in case anyone else had the same problem, I didn't mean to cause any offense.  Apologies.

**Comment by eddelbuettel** [comment]
Even if that were true, _who cares_ ?  Ubuntu 16.04 is deprecated.  Your assumption of "I can install any version of anything on any version I like" is not valid.

**Comment by eddelbuettel** [comment]
If you (or your friend) insists on an outdated version of the OS, it is _your problem_ to potentially have to stick with matching-to-the-old-software versions of what you want to use.

And I just installed `RcppSMC` fine from source on an Ubuntu 16.04 "Xenial" docker instance.

**Comment by sgmshaik** [comment]
Thank you for your help and advice. And apologies for any inconvenience. It's just me and no friend.

**Comment by eddelbuettel** [comment]
No worries.

Ubuntu is great.  But _please please please_ do not fear the upgrade.  _Do_ move to 18.04 at least.  I have been updating every six months to every now release -- typically waiting a month or two.  Currently at 18.10, it will be 19.04 (which comes out this month) for me by May or June.  

Totally worth it as you avoid all these types of headaches.  Ask questions for R releated things on r-sig-debian.  Cheers.

**Comment by eddelbuettel** [comment]
For the record:

```sh
root> R -e 'install.packages("RcppSMC", repos="https://cloud.r-project.org")'
R -e 'install.packages("RcppSMC", repos="https://cloud.r-project.org")'

R version 3.2.3 (2015-12-10) -- "Wooden Christmas-Tree"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("RcppSMC", repos="https://cloud.r-project.org")
Installing package into '/usr/local/lib/R/site-library'
(as 'lib' is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/RcppSMC_0.2.1.tar.gz'
Content type 'application/x-gzip' length 50298 bytes (49 KB)
==================================================
downloaded 49 KB

* installing *source* package 'RcppSMC' ...
** package 'RcppSMC' successfully unpacked and MD5 sums checked
** libs
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c LinReg.cpp -o LinReg.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c LinReg_LA.cpp -o LinReg_LA.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c LinReg_LA_adapt.cpp -o LinReg_LA_adapt.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppExports.cpp -o RcppExports.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blockpfgaussianopt.cpp -o blockpfgaussianopt.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c nonLinPMMH.cpp -o nonLinPMMH.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pflineart.cpp -o pflineart.o
g++ -I/usr/share/R/include -DNDEBUG -I../inst/include  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/RcppArmadillo/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pfnonlinbs.cpp -o pfnonlinbs.o
g++ -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o RcppSMC.so LinReg.o LinReg_LA.o LinReg_LA_adapt.o RcppExports.o blockpfgaussianopt.o nonLinPMMH.o pflineart.o pfnonlinbs.o -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/RcppSMC/libs
** R
** data
*** moving datasets to lazyload DB
** inst
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (RcppSMC)

The downloaded source packages are in
	'/tmp/Rtmpkmitup/downloaded_packages'
> 
> 
root> 
```

and

```sh
root> COLUMNS=-90 dpkg -l 'r-cran-rcpp*'
COLUMNS=-90 dpkg -l 'r-cran-rcpp*'
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                 Version         Architecture    Description
+++-====================-===============-===============-=============================================
ii  r-cran-rcpp          0.12.3-1        amd64           GNU R package for Seamless R and C++ Integrat
ii  r-cran-rcpparmadillo 0.6.500.4.0-1   amd64           GNU R package for Armadillo C++ linear algebr
root> 
```

**Comment by eddelbuettel** [comment]
Oh, and lastly:  This was never an issue in `RcppSMC`.  If anything you encounter "cannot install _current_ `RcppArmadillo` from source on _acnient_ Ubuntu" which is really a completely different issue.

**Comment by sgmshaik** [comment]
Yes, thanks I am uprgrading ubuntu on my desktop now and thanks for the advice regarding r-sig-debian and the prompt replies I really appreciate it. I am new to R. Thanks for the awesome SMC package. looking forward to playing around with it.

**Comment by eddelbuettel** [comment]
Sounds good!

**Comment by eddelbuettel** [comment]
And not to beat a dead horse, but trying to help here -- when in Ubuntu 16.04 you try to install `RcppArmadillo` you see

```r
> install.packages("RcppArmadillo", repos="http://cloud.r-project.org")
Installing package into '/usr/local/lib/R/site-library'
(as 'lib' is unspecified)
Warning message:
package 'RcppArmadillo' is not available (for R version 3.2.3) 
> 
```

because the default version of R there is R 3.2.3 -- three years old. One of the dependencies will have a versioned depends for something newer.  So either
-  if you installed a newer R -- as you
can _e.g._ via CRAN's mirror of our ported R binaries -- it would work.,
- or a s would of course _installing a binary_ via `sudo apt-get install r-cran-rcpparmadillo`.

As always, there are different routes to the eventual goal.
